Docs: add dev blog post for client breaking changes in 12517` (#2467)
input-output-hk/mithril: docs/website/blog/2025-05-05-distribution-2517.md

### Distribution `2517` is now available

:::info Update 2025/05/06

> We have released the hotfix `2517.1` distribution that fixes a bug in the the `2517.0` distribution which caused the Cardano node to fail during startup when using a snapshot downloaded with the Mithril client.

Reviewer: ```suggestion
The team released the hotfix `2517.1` to address a bug in the `2517.0` distribution that caused the Cardano node to fail during startup when using a snapshot downloaded with the Mithril client.
```
